### DISCUSSION TITLE
How to close ui.notify(type='ongoing’)?

### DISCUSSION THREAD

**Discussion by JS-Aibel** [zauberzeug/nicegui#890] [Q&A] [answered]
### Description

`ui.notify(type='ongoing’)` never disappears.
Is there a way to programmatically close the ui.notify type ongoing? For example by change it to another type, or trigger the close button (closeBtn) somehow? The use case for using the ui.notify ongoing, is to show the message until something is finished.

```py
from nicegui import ui

ui.button('Click', on_click=lambda: ui.notify('Waiting!', type='ongoing'))

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
This is similar to discussion https://github.com/zauberzeug/nicegui/discussions/515.
Unfortunately, we still don't have a solution. Any suggestions or PRs are very welcome.

**Comment by aljosahafner** [comment]
Sorry for writing into an already closed thread, but it would be extremely useful to provide a way to close an ongoing notification automatically from the code. Example use case is notifying the user whether a long-running background function is actually running and when it stops...
If there's already a solution I'd appreciate some guidance on how to implement it. Thanks in advance.

**Reply by falkoschindler** [reply]
Currently `ui.notify` is pretty limited. But I'm working on a new implementation: PR https://github.com/zauberzeug/nicegui/pull/2031. This should resolve your use case.

**Reply by aljosahafner** [reply]
Thanks a lot for the answer. I will keep an eye on when this is ready!
